File: skimmer/mergeDataSets.cc
/*
Executable C++ file that calls the function mergeAndRemoveOverlap from the framework
*/
// typical use case: merging datasets together with potentially overlapping events


// include c++ library classes 
#include <string>
#include <vector>
#include <exception>
#include <iostream>

// inlcude function to call
#include "../Tools/interface/mergeAndRemoveOverlap.h"

int main( int argc, char* argv[] ){
    std::cerr << "###starting###" << std::endl;
    
    if( argc < 4 ){
        std::cerr << "ERROR: need more arguments to run: " << std::endl;
        std::cerr << " - output file name" << std::endl;
	std::cerr << " - at least two input files" << std::endl;
	std::cerr << "(similar command line structure as hadd)" << std::endl;
        return -1;
    }
    
    // parse arguments
    std::vector< std::string > argvStr( &argv[0], &argv[0] + argc );
    std::string& output_file_path = argvStr[1];
    std::vector<std::string> input_files;
    for(int i=2; i<argc; i++){
	input_files.push_back(argvStr[i]);
    }

    mergeAndRemoveOverlap(input_files, output_file_path, false);
    std::cerr << "###done###" << std::endl;
}

File: skimmer/mergeDataSets.py
#################################
# a submitter for mergeDataSets #
#################################

import sys
import os
sys.path.append('../jobSubmission')
import condorTools as ct

if __name__=='__main__':

    # read command line args
    if len(sys.argv)<4:
	print('ERROR: need at least three command line arguments:')
	print('- target file')
	print('- at least two files to merge')
	sys.exit()

    targetfile = sys.argv[1]
    inputfiles = sys.argv[2:]
    exe = './mergeDataSets'
    
    # check if input files exist
    for f in inputfiles:
	if not os.path.exists(f):
	    raise Exception('ERROR: input file {} does not exist'.format(f))
    
    # check if target file exists
    if os.path.exists(targetfile):
	raise Exception('ERROR: target file {} already exist,'.format(targetfile)
			+' please remove it manually before recreating')

    # check if executable exists
    if not os.path.exists(exe):
	raise Exception('ERROR: executable {} does not exist'.format(exe))

    # print and ask for confirmation
    print('will merge the following files:')
    for f in inputfiles: print('- {}'.format(f))
    print('into the target file {}'.format(targetfile))
    print('continue? (y/n)')
    go = raw_input()
    if go != 'y': sys.exit()

    # make the command
    cmd = '{} {}'.format(exe,targetfile)
    for f in inputfiles: cmd += ' {}'.format(f)

    # submit the command
    ct.submitCommandAsCondorJob( 'cjob_mergeDataSets', cmd )
